crypto/merklesignature/keysBuilder.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package merklesignature
+
+import (
+	"runtime"
+	"sync"
+	"sync/atomic"
+
+	"github.com/algorand/go-algorand/crypto"
+)
+
+// KeysBuilder Responsible for generate slice of falcon keys
+func KeysBuilder(numberOfKeys uint64) ([]crypto.FalconSigner, error) {
+	numOfKeysPerRoutine, _ := calculateRanges(numberOfKeys)
+
+	var terminate int64
```

Comment:
nit: `context.WithCancel` could be used instead of terminate, it's probably more idiomatic than the integer / atomic ops pattern here.

Reply:
done